### ISSUE TITLE
After using torch.load(), I can't get exactly the same AUC and testing loss as they were during training. 

### ISSUE THREAD

**Issue by WenOOI** [SCLBD/DeepfakeBench#36]
After using torch.load(), I can't get exactly the same AUC and testing loss as they were during training. 
I executed the test task using your code, which fixed the random seed and utilized model.eval(). However, I still received slightly different results.
`
def init_seed(config):
    if config['manualSeed'] is None:
        config['manualSeed'] = random.randint(1, 10000)
    random.seed(config['manualSeed'])
    torch.manual_seed(config['manualSeed'])
    if config['cuda']:
        torch.cuda.manual_seed_all(config['manualSeed'])
`
The log during training.
`
2023-11-12 10:39:09,987 - INFO - dataset: FaceForensics++    step: 2287    testing-loss, overall: 0.3616461753845215    
2023-11-12 10:39:09,987 - INFO - dataset: FaceForensics++    step: 2287    testing-metric, acc: 0    testing-metric, auc: 0.9837027633520393    testing-metric, eer: 0.06142668428005284    testing-metric, ap: 0.9912653159421054    
2023-11-12 10:39:09,988 - INFO - ===> Test Done!
`
The log during testing.
`
dataset: FaceForensics++        testing-metric, acc: 0    testing-metric, auc: 0.9547929684664158    testing-metric, eer: 0.10706303084488154    testing-metric, ap: 0.9871193988897207    testing-metric, pred: [9.9990773e-01 9.9993742e-01 1.3892369e-04 ... 7.7735519e-01 9.9993634e-01
 9.9982518e-01]    testing-metric, label: [1 1 0 ... 1 1 0]    
dataset: FaceForensics++        testing-metric, acc: 0    testing-metric, auc: 0.9547929684664158    testing-metric, eer: 0.10706303084488154    testing-metric, ap: 0.9871193988897207    testing-metric, pred: [9.9990773e-01 9.9993742e-01 1.3892369e-04 ... 7.7735519e-01 9.9993634e-01
 9.9982518e-01]    testing-metric, label: [1 1 0 ... 1 1 0]    
===> Test Done!
`

**Comment by WenOOI** [comment]
I attempted to annotate the loss.backward() function in train_epoch to isolate any differences in code, but obtain inconsistent outcomes.

But it is worth noting that if I train 100 iterations after torch. load, I will get similar results.

**Comment by WenOOI** [comment]
By the way, I set 'strict=True' when using 'torch.load()'.

**Comment by YZY-stack** [comment]
Hi, I guess the reason causing the discrepancies you're encountering might stem from inconsistencies in the data collection process for the testing data. I recommend using my latest code update to train your model again and then test it using the evaluation script provided. Hopefully, this resolves the issue. Thank you and good luck.

**Comment by WenOOI** [comment]
Thanks for your help! After using the latest code, the issue is solved.
